core/src/main/java/io/kestra/core/models/triggers/Trigger.java

```diff
@@ -214,11 +217,11 @@ public static Trigger of(Flow flow, AbstractTrigger abstractTrigger, ConditionCo
             .nextExecutionDate(((PollingTriggerInterface) abstractTrigger).nextEvaluationDate(conditionContext, Optional.empty()))
             .stopAfter(abstractTrigger.getStopAfter())
             .disabled(lastTrigger.map(TriggerContext::getDisabled).orElse(Boolean.FALSE))
-            .backfill(lastTrigger.map(TriggerContext::getBackfill).orElse(null))
+            .backfill(null)
             .build();
     }
 
-    public static Trigger update(Trigger currentTrigger, Trigger newTrigger) {
+    public static Trigger update(Trigger currentTrigger, Trigger newTrigger, AbstractTrigger abstractTrigger, ConditionContext conditionContext) throws Exception {
```

Comment:
I think it's better to calculate the next date out of this method so any exception can be handled easily.
Those methods inside the Trigger object are designed to be builders/mappers.